web/permissions/permissions.go

```diff
@@ -129,10 +138,86 @@ func listPermissions(c echo.Context) error {
 	return json.NewEncoder(resp).Encode(doc)
 }
 
+func patchPermission(c echo.Context) error {
+	instance := middlewares.GetInstance(c)
+	current, err := getPermission(c)
+	if err != nil {
+		return err
+	}
+
+	var patch permissions.Permission
+	if _, err = jsonapi.Bind(c.Request(), &patch); err != nil {
+		return err
+	}
+
+	patchSet := patch.Permissions != nil && len(*patch.Permissions) > 0
```

Comment:
Just a question: why `patch.Permissions` is a pointer to a slice, and not just a slice? It's not common to need a pointer to a slice, as a slice is already a kind of pointer.